Extensions/Optimizely/Src/Tasks/OptimizelyX/task.json

```diff
@@ -0,0 +1,157 @@
+            "label": "Optimizely-X Service Connection",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "Select the Optimizely-X service endpoint to be used to be used to manage A/B test. [Optimizely-X endpoint](https://aka.ms/)"
+        },
+        {
+            "name": "Action",
+            "type": "pickList",
+            "label": "Action",
+            "defaultValue": "StartAbTest",
+            "required": true,
+            "options": {
+                "StartAbTest": "Start A/B test",
+                "PauseAbTest": "Pause A/B test"
+            },
+            "helpMarkDown": "Specify the action to be performed."
+        },
+        {
+            "name": "Project",
+            "type": "pickList",
+            "label": "Project",
+            "defaultValue": "",
+            "required": true,
+            "properties": {
+                "EditableOptions": true
+            },
+            "helpMarkDown": "Select the name of the project."
+        },
+        {
+            "name": "Experiment",
+            "type": "pickList",
+            "label": "Experiment",
+            "defaultValue": "",
+            "required": true,
+            "properties": {
+                "EditableOptions": true
+            },
+            "helpMarkDown": "Select the name of the A/B test."
+        },
+        {
+            "name": "Traffic",
+            "type": "string",
+            "label": "Traffic",
+            "defaultValue": "100",
+            "required": true,
+            "groupName": "StartAbTestInputGroup",
+            "helpMarkDown": "Specify what percentage of total traffic you would like to A/B test."
+        },
+        {
+            "name": "TrafficByVariation",
+            "type": "multiLine",
+            "label": "Experiment variations",
+            "defaultValue": "",
+            "required": true,
+            "groupName": "StartAbTestInputGroup",
+            "properties": {
+                "editorExtension": "ms.vss-services-azure.parameters-grid"
+            },
+            "helpMarkDown": "Specify variations for the experiment and their corresponding traffic percentages."
+        },
+        {
+            "name": "Audience",
+            "type": "pickList",
+            "label": "Audience",
+            "defaultValue": "",
+            "required": false,
+            "groupName": "StartAbTestInputGroup",
+            "properties": {
+                "MultiSelect": "True",
+                "DisableManageLink": "True",
+                "EditableOptions": "True"
+            },
+            "helpMarkDown": "Specify the Target Audience for this A/B test."
+        }
+    ],
+    "dataSourceBindings": [
+        {
+            "endpointId": "$(OptimizelyXEndpoint)",
+            "target": "Project",
+            "dataSourceName": "OptimizelyXProjects",
+            "resultTemplate": "{ \"Value\" : \"{{{id}}}\", \"DisplayValue\" : \"{{{name}}}\" }"
```

Comment:
Agree, but problem with this approach is that experiment data source call take project id as input. if logic being changed to take name as input instead of id then experiment's data source call won't work.